Describe Narayana's BOMs usage and function

marcosgopen opened the pull request:
NO_TEST

This PR addresses Laird's forum question in 'narayana-users' conversation "[Why does the Narayana BOM feature third-party dependencies?](https://groups.google.com/g/narayana-users/c/CCWr8ZBIe_Q)"
I added comment for Narayana BOMs in order to clarify their function

jmfinelli commented on narayana-bom/pom.xml:
```diff
@@ -16,6 +16,13 @@
+    In this way when a dependency is updated it is updated in all Narayana repositories
+    and we could immediately test the change everywhere and evaluate its impact 
+-->
+
```
Thanks Marco. WDYT if we go for something shorter? Something like:
`This Bill of Materials (BOM) is for centralising dependency management across the various modules of Narayana.`

marcosgopen replied:
Yes, I think something shorter would be better. I would underline anyway that this is for " developers' use across the Narayana repos rather than end users'." What about keeping that too?

marcosgopen commented:
@jmfinelli I updated the comment, what about it now?

mmusgrov reviewed (changes requested):
@marcosgopen If the purpose is to address Laird's forum question then you should update the <description> tag.

mmusgrov commented on narayana-bom-test/pom.xml:
```diff
@@ -17,6 +17,11 @@
 
+<!--
+	This Bill of Materials (BOM) is for centralising dependency management across the various modules of Narayana.
+    The BOM is intended for development purposes within Narayana. End users should not use it for integrating Narayana into their projects.
```
Why should end users not use it?

marcosgopen replied:
Thanks for you review @mmusgrov , I removed the last sentence because I agree it has a different meaning than I intended. My point was that the BOM is intended for development purpose, so the previous sentence should be enough.

mmusgrov replied:
And users should still be okay to use it. If they want to use different versions of a dependency then they use the `<exclusions>` tag.

marcosgopen replied:
Yes, I agree.

marcosgopen commented:
> @marcosgopen If the purpose is to address Laird's forum question then you should update the tag.

Thanks @mmusgrov for the review, I updated the PR description. If needed I will update the commit message too.

mmusgrov commented:
> > @marcosgopen If the purpose is to address Laird's forum question then you should update the tag.
> 
> Thanks @mmusgrov for the review, I updated the PR description. If needed I will update the commit message too.

I meant the `<description>` tag of the pom

marcosgopen commented:
> > > @marcosgopen If the purpose is to address Laird's forum question then you should update the tag.
> > 
> > 
> > Thanks @mmusgrov for the review, I updated the PR description. If needed I will update the commit message too.
> 
> I meant the `<description>` tag of the pom

Very nice point! I updated the pom's description tag.
